scripts/generate-sitemap.js

```diff
@@ -13,63 +13,147 @@ async function readSchemaJson(filePath) {
   }
 }
 
-async function processDirectory(dirPath, relativePath = "") {
-  const entries = await fs.readdir(dirPath, { withFileTypes: true });
-  let paths = [];
-
-  for (const entry of entries) {
-    const entryPath = path.join(dirPath, entry.name);
-
-    if (entry.name === "index.json") {
-      continue;
+async function processFile(fullPath, relativePath, name) {
+  const schemaPath = _path.join(fullPath, name);
+  const schemaData = await readSchemaJson(schemaPath);
+  if (!schemaData) {
+    return null;
+  }
+  const pageName = name.split(".")[0];
+  const permalink = _path.join(relativePath, pageName);
+  const siteMapEntry = {
+    permalink,
+    title:
+      schemaData.page.title ||
+      pageName.charAt(0).toUpperCase() + pageName.slice(1),
+  };
+  // check if file is actually an index page for a directory
+  try {
+    const directoryPath = _path.join(fullPath, name.split(".")[0]);
+    const stats = await fs.stat(directoryPath);
+    if (stats.isDirectory()) {
+      return {
+        ...siteMapEntry,
+        children: await processDirectory(directoryPath, permalink),
+      };
     }
+    return siteMapEntry;
+  } catch (error) {
+    return siteMapEntry;
+  }
```

Comment:
nit: for sanity can we log out the error (with json stringify) to help with potential debugging?

Reply:
[0f106f2](https://github.com/isomerpages/isomer-next-build/pull/1/commits/0f106f241ad6630097a0e728aaea168361d6ee2a)

refactored this jank code (was relying on the try catch to see if the path exists), using `existsSync` instead